install/autodetect.go

```diff
@@ -166,5 +168,33 @@ func (k *K8sInstaller) autodetectAndValidate(ctx context.Context) error {
 		return fmt.Errorf("invalid encryption mode")
 	}
 
+	k.autodetectKubeProxy(ctx)
+	return nil
+}
+
+func (k *K8sInstaller) autodetectKubeProxy(ctx context.Context) error {
+	kubeSysNameSpace := "kube-system"
+	dsList, err := k.client.ListDaemonSet(ctx, kubeSysNameSpace, metav1.ListOptions{})
+	if err != nil {
+		k.Log("⏭️ Skipping auto kube-proxy detction")
+		return nil
+	}
+
+	for _, ds := range dsList.Items {
+		if strings.Contains(ds.Name, "kube-proxy") {
```

Comment:
maybe we should use `if ds.Nmae == "kube-proxy"` instead of `strings.Contains`?

Reply:
thanks for the suggestion. Yes, the Kube-proxy auto-detection may take a different DaemonSet as Kube-Proxy if they also contain "kube-proxy". Just for curiosity, have you found such cases like that? @brb wdyt?

Reply:
I've only seen "kube-proxy", so the suggested change makes sense to me.

Reply:
Thanks @brb. If you think it's to better to change now, I can create a git Issue for that? (I can fix that too)

Reply:
I think it faster to create a PR than an issue, so I'd go with a fix. Thanks.